apps/opik-backend/src/main/java/com/comet/opik/domain/SpanDAO.java

```diff
@@ -892,10 +892,18 @@ public Mono<Long> partialInsert(@NonNull UUID id, @NonNull UUID projectId, @NonN
                 .reduce(0L, Long::sum);
     }
 
-    private Publisher<? extends Result> update(UUID id, SpanUpdate spanUpdate, Connection connection) {
+    private Publisher<? extends Result> update(UUID id, SpanUpdate spanUpdate, Connection connection, Span existingSpan) {
+        if (spanUpdate.model() != null || spanUpdate.usage() != null) {
+            spanUpdate = spanUpdate.toBuilder()
+                    .model(spanUpdate.model() != null ? spanUpdate.model() : existingSpan.model())
+                    .usage(spanUpdate.usage() != null ? spanUpdate.usage() : existingSpan.usage())
+                    .build();
```

Comment:
Can we use the approach we were using before? Select inserts, in this case, are less prone to data lost due to concurrent updates

Reply:
I'm using this approach partially. Reason why I can't use it fully, is that I must calculate new cost in Java before inserting it, as discussed we should not move cost calculation to Clickhouse query due to potential complexity in future.